src/server/internalServer.cpp

```diff
@@ -878,4 +881,61 @@ std::unique_ptr<Response> InternalServer::handle_content(const RequestContext& r
   }
 }
 
+
+std::unique_ptr<Response> InternalServer::handle_raw(const RequestContext& request)
+{
+  if (m_verbose.load()) {
+    printf("** running handle_raw\n");
+  }
+
+  std::string bookName;
+  std::string kind;
+  try {
+     bookName = request.get_url_part(1);
+     kind = request.get_url_part(2);
+  } catch (const std::out_of_range& e) {
+     return Response::build_404(*this, request.get_full_url(), bookName, "", "");
+  }
+
+  if (kind != "meta" && kind!= "content") {
+    const std::string error_details = kind + " is not a valid request for raw content.";
+    return Response::build_404(*this, request.get_full_url(), bookName, "", "");
+  }
+
+  std::shared_ptr<zim::Archive> archive;
+  try {
+    const std::string bookId = mp_nameMapper->getIdForName(bookName);
+    archive = mp_library->getArchiveById(bookId);
+  } catch (const std::out_of_range& e) {}
+
+  if (archive == nullptr) {
+    const std::string error_details = "No such book: " + bookName;
+    return Response::build_404(*this, request.get_full_url(), bookName, "", error_details);
+  }
+
+  auto itemPath = request.get_url().substr(bookName.size()+kind.size()+6);
```

Comment:
6 is a magic number here. Please link it to `"/raw/" + "/"` in a comment .

Reply:
Done (and fixed the value)